jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/InputStreamResponseListener.java

```diff
@@ -187,6 +189,12 @@ public void onComplete(Result result)
         chunkCallbacks.forEach(t -> t.releaseAndFail(failure));
     }
 
+    @Override
+    public void close() throws IOException
+    {
+        stream.updateAndGet(input -> input == null ? InputStream.nullInputStream() : input).close();
```

Comment:
This doesn't really change anything if the stream was never created.

I expected this to release all chunks that have been received already, and probably abort an already demanded but not yet received chunk, or maybe the whole response (does that actually happen in `Input.close`?).

So when the InputStream hasn't been created yet, the behavior should be similar to `getInputStream().close()`. You could achieve this by changing this slightly to create a real `Input` instead of a `nullInputStream`, or you could try avoiding to create a stream and invoking the corresponding logic directly.

Reply:
@mperktold good catch!